vpr/src/base/SetupVPR.cpp

```diff
@@ -109,11 +110,20 @@ void SetupVPR(const t_options* Options,
 
     if (readArchFile == true) {
         vtr::ScopedStartFinishTimer t("Loading Architecture Description");
-        XmlReadArch(Options->ArchFile.value().c_str(),
-                    TimingEnabled,
-                    Arch,
-                    device_ctx.physical_tile_types,
-                    device_ctx.logical_block_types);
+        if (Options->FPGAInterchangeDevice) {
+            FPGAInterchangeReadArch(Options->FPGAInterchangeDeviceFile().value().c_str(),
+                                    TimingEnabled,
+                                    Arch,
+                                    device_ctx.physical_tile_types,
+                                    device_ctx.logical_block_types);
+            VTR_LOG("Use FPGA Interchange device\n");
```

Comment:
Should we show the log before calling the ReadArch() function? It will be easier for debugging

Reply:
Done